### PR TITLE
Adding ability to delete a IIIF collection

### PR DESCRIPTION
**Pull request by JackLewis-digirati**
This PR modifies `deleteCollection` so that deleting a IIIF collection will also remove the associated record from S3

### CODE REVIEW

File: src/IIIFPresentation/API.Tests/Integration/DeleteCollectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.Forbidden);
+    }
+
+    [Fact]
+    public async Task DeleteCollection_DeletesCollection_WhenAllValuesProvided()
+    {
+        // Arrange
+        var initialCollection = new Collection()
```
Comment (donaldgray) [inline src/IIIFPresentation/API.Tests/Integration/DeleteCollectionTests.cs:78]: There's a lot of repetition in Collection tests - could we use `DatabaseTestDataPopulation.AddTestCollection()` here (may need tweaked)? It'll keep the tests more concise to show what's useful.

File: src/IIIFPresentation/API.Tests/Integration/DeleteCollectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        errorResponse!.ErrorTypeUri.Should()
+            .Be("http://localhost/errors/DeleteCollectionType/CannotDeleteRootCollection");
+        errorResponse.Detail.Should().Be("Cannot delete a root collection");
+    }
+
+
+    [Fact]
+    public async Task DeleteCollection_FailsToDeleteCollection_WhenAttemptingToDeleteRootDirectly()
```
Comment (donaldgray) [inline src/IIIFPresentation/API.Tests/Integration/DeleteCollectionTests.cs:159]: This test looks identical to `DeleteCollection_FailsToDeleteCollection_WhenAttemptingToDeleteRoot()`

File: src/IIIFPresentation/API/Infrastructure/AWS/IIIFS3Service.cs [hunk 3]
```diff
@@ -51,6 +51,16 @@ public async Task SaveIIIFToS3(ResourceBase iiifResource, IHierarchyResource dbR
...
     }
+    
+    /// <summary>
+    /// Delete IIIF resource from S3
+    /// </summary>
+    public void DeleteIIIFFromS3(IHierarchyResource dbResource)
+    {
+        logger.LogDebug("Deleting resource {Customer}:{ResourceId} file from S3", dbResource.CustomerId, dbResource.Id);
+        var item = new ObjectInBucket(options.CurrentValue.AWS.S3.StorageBucket, dbResource.GetResourceBucketKey());
+        bucketWriter.DeleteFromBucket(item);
```
Comment (donaldgray) [inline src/IIIFPresentation/API/Infrastructure/AWS/IIIFS3Service.cs:62]: unawaited `Task`

File: src/IIIFPresentation/API.Tests/Integration/DeleteCollectionTests.cs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+            },
+            Parent = RootCollection.Id,
+            Slug = slug
+        };
+
+        var requestMessage =
+            HttpRequestMessageBuilder.GetPrivateRequest(HttpMethod.Post, $"{Customer}/collections",
+                collection.AsJson());
+
+        var response = await httpClient.AsCustomer(1).SendAsync(requestMessage);
```
Comment (griffri) [inline src/IIIFPresentation/API.Tests/Integration/DeleteCollectionTests.cs:214]: It looks like all of these tests make requests as customer 1 - would it be worth defining a `customerId` const?

**Review by griffri — APPROVED** [review]

**Review by donaldgray — CHANGES_REQUESTED** [review]

**Review by donaldgray — APPROVED** [review]

**Pull request merged**